source/common/http/http1/codec_impl.cc

```diff
@@ -944,7 +944,7 @@ void ClientConnectionImpl::onMessageComplete(HeaderMapImplPtr&& trailers) {
     // reused, unwind any outstanding readDisable() calls here. Only do this if there are no
     // pipelined responses remaining. Also do this before we dispatch end_stream in case the caller
     // immediately reuses the connection.
-    if (pending_responses_.empty()) {
+    if (connection_.state() == Network::Connection::State::Open && pending_responses_.empty()) {
```

Comment:
I'd be inclined to move this up and not do any work  (ignore pending responses) if the connection is not open.   @mattklein123 WDYT?

Reply:
You mean moving this connection state check to https://github.com/envoyproxy/envoy/blob/e9b59142c329ca9bd32cc4ab352f617f18260d2e/source/common/http/http1/codec_impl.cc#L936 ?

There are some calls to the decoder below that I think we need to perform, otherwise handling of upstream responses framed by connection closed which are proxied over H2 or chunked-encoded via HTTP/1.1 to the downstream won't be framed correctly.

Reply:
Yeah this looks right to me. What did you have in mind @alyssawilk?

Reply:
let's leave as is then